flexmeasures/cli/monitor.py

```diff
@@ -89,18 +120,159 @@ def monitor_tasks(task, custom_message):
+    """
+    user_info_list = [
+        f"{user.username} (last contact was {user.last_seen_at})" for user in users
+    ]
+
+    msg = (
+        f"The following user(s) have not contacted this FlexMeasures server for more"
+        f" than {last_seen_delta}, even though we expect they would have:\n"
+    )
+    for user_info in user_info_list:
+        msg += f"\n- {user_info}"
+
+    # Sentry
+    set_sentry_context(
+        "last_seen_context",
+        {
+            "delta": last_seen_delta,
+            "alerted_users": alerted_users,
+            "account_role": account_role,
+            "user_role": user_role,
+        },
+    )
+    capture_message_for_sentry(msg)
+
+    # Email
+    msg += "\n"
+    if account_role:
+        msg += f"\nThis alert concerns users whose accounts have the role '{account_role}'."
+    if user_role:
+        msg += f"\nThis alert concerns users who have the role '{user_role}'."
+    if alerted_users:
+        msg += "\n\nThe user(s) has/have been notified as well."
+    else:
+        msg += (
+            "\n\nThe user(s) has/have not been notified (--alert-users was turned off)."
+        )
+    email_recipients = app.config.get("FLEXMEASURES_MONITORING_MAIL_RECIPIENTS", [])
+    if len(email_recipients) > 0:
+        email = Message(
+            subject="Last contact by user(s) too long ago", bcc=email_recipients
+        )
+        email.body = msg
+        app.mail.send(email)
+
+    app.logger.error(msg)
+
+
+@fm_monitor.command("last-seen")
+@with_appcontext
+@click.option(
+    "--maximum-minutes-since-last-seen",
+    type=int,
+    required=True,
+    help="Maximal minutes since last request.",
+)
+@click.option(
+    "--alert-users",
+    type=bool,
+    default=False,
+    help="If True, also send an email to the user. Defaults to False, as these users are often bots.",
+)
+@click.option(
+    "--account-role",
+    type=str,
+    help="The name of an account role to filter for.",
+)
+@click.option(
+    "--user-role",
+    type=str,
+    help="The name of an account role to filter for.",
+)
+def monitor_last_seen(
+    maximum_minutes_since_last_seen: int,
+    alert_users: bool = False,
+    account_role: Optional[str] = None,
+    user_role: Optional[str] = None,
+):
+    """
+    Check if given users last contact (via a request) happened less than the allowed time ago.
+
+    Helpful for user accounts that are expected to contact FlexMeasures regularly (in an automated fashion).
+    If the last contact was too long ago, we send alerts via Sentry, as well as emails to monitoring mail recipients.
+    The user can be informed, as well.
+
+    The set of users can be narrowed down by roles.
+    """
+    last_seen_delta = timedelta(minutes=maximum_minutes_since_last_seen)
+
+    # find users we haven't seen in the given time window
+    users: List[User] = User.query.filter(
+        User.last_seen_at < server_now() - last_seen_delta
```

Comment:
This is comparing a timezone-naive datetime (in local utc) to a timezone-aware datetime (in the server's timezone), which would normally fail with `TypeError: can't compare offset-naive and offset-aware datetimes`. I'm not sure why it doesn't fail here, but my guess is that the expression isn't evaluated until after the `filter` function had a chance to convert `server_now()` to local utc.

```
server_now (type <class 'datetime.datetime'>): 2022-12-01 21:01:26.118386+09:00
last_login_at (type <class 'datetime.datetime'>): 2022-12-01 12:01:14.686337
last_seen_at (type <class 'datetime.datetime'>): 2022-12-01 12:01:26.071386
```

If `last_login_at` and `last_seen_at` are stored as local utc, why do we use `server_now()` instead of `datetime.utcnow()` to set them (in `models/user.py` and to compare them (here)? Seems like unnecessary conversions.

Incidentally, the docstring and logic of `server_now()` could use an update. The docstring says it localizes to the serverś timezone, but it actually converts. And the logic replaces the timezone info a couple of times (while calling two other util functions and several datetime methods). I believe this would suffice:
```
def as_server_time(dt: datetime) -> datetime:
    """The current time (timezone aware), converted to the timezone of the FlexMeasures platform."""
    return get_timezone().fromutc(datetime.utcnow())
```

Reply:
Great - simpler is better!